packages/dds/tree/src/test/shared-tree/opSize.bench.ts

```diff
@@ -0,0 +1,1135 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import Table from "easy-table";
+import { isInPerformanceTestingMode } from "@fluid-tools/benchmark";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { emptyField, FieldKinds, namedTreeSchema, singleTextCursor } from "../../feature-libraries";
+import { ISharedTree } from "../../shared-tree";
+import { brand } from "../../util";
+import { ITestTreeProvider, TestTreeProvider } from "../utils";
+import {
+    FieldKey,
+    fieldSchema,
+    JsonableTree,
+    moveToDetachedField,
+    rootFieldKey,
+    rootFieldKeySymbol,
+    SchemaData,
+    TransactionResult,
+    Value,
+    ValueSchema,
+} from "../../core";
+
+const stringSchema = namedTreeSchema({
+    name: brand("String"),
+    extraLocalFields: emptyField,
+    value: ValueSchema.String,
+});
+
+export const childSchema = namedTreeSchema({
+    name: brand("Test:Opsize-Bench-Child"),
+    localFields: {
+        data: fieldSchema(FieldKinds.value, [stringSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const parentSchema = namedTreeSchema({
+    name: brand("Test:Opsize-Bench-Root"),
+    localFields: {
+        children: fieldSchema(FieldKinds.sequence, [childSchema.name]),
+    },
+    extraLocalFields: emptyField,
+});
+
+export const rootSchema = fieldSchema(FieldKinds.value, [parentSchema.name]);
+
+export const fullSchemaData: SchemaData = {
+    treeSchema: new Map([
+        [stringSchema.name, stringSchema],
+        [parentSchema.name, parentSchema],
+    ]),
+    globalFieldSchema: new Map([[rootFieldKey, rootSchema]]),
+};
+
+const initialTestJsonTree = {
+    type: parentSchema.name,
+    fields: {
+        children: [],
+    },
+};
+
+const childrenFieldKey: FieldKey = brand("children");
+
+/*
+ * Updates the given `tree` to the given `schema` and inserts `state` as its root.
+ */
+function initializeTestTree(tree: ISharedTree, state: JsonableTree = initialTestJsonTree) {
+    tree.storedSchema.update(fullSchemaData);
+    // inserts a node with the initial AppState as the root of the tree
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor(state);
+        const field = editor.sequenceField(undefined, rootFieldKeySymbol);
+        field.insert(0, writeCursor);
+        return TransactionResult.Apply;
+    });
+}
+
+const getJsonNode = (desiredByteSize: number): JsonableTree => {
+    const node = {
+        type: childSchema.name,
+        fields: {
+            data: [{ value: "", type: stringSchema.name }],
+        },
+    };
+
+    let nodeByteSize = new TextEncoder().encode(JSON.stringify(node)).length;
+
+    const sizeIncrementor = "a"; // 1 byte
+    const incrementorByteSize = new TextEncoder().encode(sizeIncrementor).length;
+
+    while (nodeByteSize < desiredByteSize) {
```

Comment:
Instead of doing it this way, can we do `desiredByteSize - nodeByteSize` and then set the value using `'a'.repeat(remainingByteSize`)?

Reply:
Done in most recent commit.